include/dem/particle_particle_linear_force.h

```diff
@@ -0,0 +1,128 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2019 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Shahab Golshan, Polytechnique Montreal, 2019
+ */
+
+#include <dem/dem_solver_parameters.h>
+#include <dem/particle_particle_contact_force.h>
+#include <dem/particle_particle_contact_info_struct.h>
+
+#include <deal.II/particles/particle.h>
+#include <deal.II/particles/particle_iterator.h>
+
+#include <math.h>
+
+#include <iostream>
+#include <vector>
+
+using namespace dealii;
+
+#ifndef particle_particle_linear_force_h
+#  define particle_particle_linear_force_h
+
+/**
+ * Calculation of the linear particle-particle contact force using the
+ * information obtained from the fine search and physical properties of
+ * particles
+ *
+ * @author Shahab Golshan, Bruno Blais, Polytechnique Montreal 2019-
+ */
+
+template <int dim>
+class ParticleParticleLinearForce : public ParticleParticleContactForce<dim>
+{
```

Comment:
I'm not sure I understand, I thought your idea was to templatize the force model for the rolling friction?